src/Workspaces/Remote/Core/SolutionAssetStorage.cs

```diff
@@ -50,18 +50,18 @@ public Scope GetScope(Checksum solutionChecksum)
     /// Adds given snapshot into the storage. This snapshot will be available within the returned <see cref="Scope"/>.
     /// </summary>
     public ValueTask<Scope> StoreAssetsAsync(Solution solution, CancellationToken cancellationToken)
-        => StoreAssetsAsync(solution.State, cancellationToken);
+        => StoreAssetsAsync(solution.CompilationState, cancellationToken);
 
     /// <inheritdoc cref="StoreAssetsAsync(Solution, CancellationToken)"/>
     public ValueTask<Scope> StoreAssetsAsync(Project project, CancellationToken cancellationToken)
-        => StoreAssetsAsync(project.Solution.State, project.Id, cancellationToken);
+        => StoreAssetsAsync(project.Solution.CompilationState, project.Id, cancellationToken);
 
     /// <inheritdoc cref="StoreAssetsAsync(Solution, CancellationToken)"/>
-    public ValueTask<Scope> StoreAssetsAsync(SolutionState solution, CancellationToken cancellationToken)
+    public ValueTask<Scope> StoreAssetsAsync(SolutionCompilationState solution, CancellationToken cancellationToken)
```

Comment:
Consider further renames, since these two StoreAssetsAsync both take the same types, but have very different parameter names.

Reply:
tracking.